homeassistant/components/plex/server.py

```diff
@@ -187,7 +187,7 @@ def _update_plexdirect_hostname():
                 while error and not isinstance(error, ssl.SSLCertVerificationError):
                     error = error.__context__
                 if isinstance(error, ssl.SSLCertVerificationError):
-                    domain = urlparse(self._url).netloc.split(":")[0]
+                    domain = urlparse(self._url).netloc.partition(":")[0]
```

Comment:
How did this ever work?

```
>>> urlparse("https://user:password@google.com/abcdef")
ParseResult(scheme='https', netloc='user:password@google.com', path='/abcdef', params='', query='', fragment='')
```

What is desired at least according to spec is to split at the @ spot.  Does the Plex integration do something funny with URLs that would cause `urlparse` to give back a colon but no at sign?

Reply:
A Plex URL would never have basic auth parameters, but it could have a port (`myplexserver:1234`).

Reply:
OK.